### PR TITLE
Fix pretty_printer to print omitted type `_` marker

### LINKED ISSUE #31073: pretty printer doesn't know how to print the omitted type `_` marker
pretty printer doesn't know how to print the omitted type `_` marker

example:

```rust
fn main() {
    fn f1(x: i32, y: i32) -> i32 { y }
    let f: fn(_, i32) -> i32 = f1;
    //        ~
    //        |
    //        +--- this is pretty-printed badly.
    f(13);
}
```

pretty prints this way:

```
% ./x86_64-apple-darwin/stage1/bin/rustc --pretty -Z unstable-options pp-1.rs
fn main() {
    fn f1(x: i32, y: i32) -> i32 { y }
    let f: fn(, i32) -> i32 = f1;
    //        ~
    //        |
    //        +--- this is pretty-printed badly.
    f(13);
}
% 
```

### PR DESCRIPTION
**Pull request by nxnfufunezn**
Fixes #31073 
r? @eddyb

### CODE REVIEW

File: src/librustc_front/print/pprust.rs [hunk 1]
```diff
@@ -1935,7 +1935,7 @@ impl<'a> State<'a> {
 
     pub fn print_fn_args(&mut self,
                          decl: &hir::FnDecl,
-                         opt_explicit_self: Option<&hir::ExplicitSelf_>)
+                         opt_explicit_self: Option<&hir::ExplicitSelf_>, is_closure: bool)
```
Comment (pnkfelix) [inline src/librustc_front/print/pprust.rs]: There should be a new line separating the two arguments here

**Comment by nxnfufunezn** [comment]
@pnkfelix done

**Comment by eddyb** [comment]
This also needs a test (see `src/test/pretty` - tests with both `.rs` and `.pp`, specifically).

**Pull request merged**